go/vt/vtadmin/api.go

```diff
@@ -240,6 +216,120 @@ func (api *API) ListenAndServe() error {
 	return api.serv.ListenAndServe()
 }
 
+// ServeHTTP serves all routes matching path "/api" (see above)
```

Comment:
(This is not a blocking comment, just a high-level (and possibly hare-brained) idea.) :) 

Do you feel as though the dynamic cluster config functionality is wholly specific to PlanetScale's use case? 

Thinking longer term, one thing we could conceivably do is make vtadmin-api's service discovery and/or HTTP request handlers "plugin-able" so that Vitess operators can write their own plug-ins to handle various points in a request's lifecycle. (Put another way, vtadmin-api could expose "hooks" like, I dunno, `onRequest`, which would call out to any plugin(s) that define a handler.) 

We already take a plugin approach with the [authenticator plugin](https://github.com/vitessio/vitess/blob/f2663ec9e54c60445d77545a5387b0130ffd5762/go/vt/vtadmin/rbac/authentication.go#L137). At Slack, our authenticator plugin actually uses cookies too. :) 

The benefit to Vitess operators is that they can write and maintain their own plugins internally, without having to generalize it to where it can be added to VTAdmin. And, for the open-source project, it'd be less code for the community to maintain.

This is, of course, way out of scope for this change + would require a bunch of design work, but could be a cool approach long term. Would love to hear what you think, and if such a scheme would work for your use case.

Reply:
... Maybe I'm just thinking of custom middleware rather than "hooks", actually! lol.

Reply:
I think that would work, and it would require a lot of changes for sure :D but it would be more elegant! The reason we went with the ServeHTTP() design is so that we didn't have to change the way the `API` interface works everywhere. But it is very Planetscale-specific right now!

Trying to make this plugin-able is something I'd be v interested in in the future 😃

Reply:
Yeah! This solution is great + the way you slotted it in is very clean. :)